EasyPost/Constants.cs

```diff
@@ -136,19 +138,72 @@ public static class CarrierAccounts
             public const string UpsAccount = "UpsAccount";
 
             /// <summary>
-            ///     Carrier account types that support custom workflows.
+            ///     Carrier account types that support custom creation workflows.
             /// </summary>
-            private static List<string> CarrierTypesWithCustomWorkflows => new()
+            private static List<string> CarrierTypesWithCustomCreateWorkflows => new()
             {
                 CarrierAccountType.FedEx.Name,
                 CarrierAccountType.FedExSmartPost.Name,
                 CarrierAccountType.Ups.Name,
+                CarrierAccountType.UpsMailInnovations.Name,
+                CarrierAccountType.UpsSurePost.Name,
             };
 
-            internal static bool IsCustomWorkflowType(string carrierType) => CarrierTypesWithCustomWorkflows.Contains(carrierType);
+            /// <summary>
+            ///     Carrier account types that support custom update workflows.
+            /// </summary>
+            private static List<string> CarrierTypesWithCustomUpdateWorkflows => new()
+            {
+                CarrierAccountType.Ups.Name,
+                CarrierAccountType.UpsMailInnovations.Name,
+                CarrierAccountType.UpsSurePost.Name,
+            };
+
+            internal static string DeriveCreateEndpoint(string carrierType)
+            {
+                string endpoint = StandardCreateEndpoint;
+
+                var @switch = new SwitchCase
+                {
+                    { CarrierAccountType.FedEx.Name, () => endpoint = CustomCreateEndpoint },
+                    { CarrierAccountType.FedExSmartPost.Name, () => endpoint = CustomCreateEndpoint },
+                    { CarrierAccountType.Ups.Name, () => endpoint = UpsOAuthCreateEndpoint },
+                    { CarrierAccountType.UpsMailInnovations.Name, () => endpoint = UpsOAuthCreateEndpoint },
+                    { CarrierAccountType.UpsSurePost.Name, () => endpoint = UpsOAuthCreateEndpoint },
+                    { SwitchCaseScenario.Default, () => endpoint = StandardCreateEndpoint },
```

Comment:
suggestion: just because we can use a switch case (and I know you were really gunning for it), doesn't mean we need to 😂 this seems to be way more complicated than it needs to be because we only have 2 options. If/else works super well in those cases. Same for the update below. I want to see much less code and complexity here instead of solving for a problem that doesn't yet exist (we don't have a massive matrix of urls to account types yet).

Reply:
Improved